commit/plugin_functions.go

```diff
@@ -628,6 +630,40 @@ func validateMerkleRootsState(
 	return true, nil
 }
 
+func backgroundReaderSync(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	lggr logger.Logger,
+	reader cciptypes.CCIPReader,
+	syncTimeout time.Duration,
+	ticker <-chan time.Time,
+) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				wg.Done()
+				return
+			case <-ticker:
+				timeoutCtx, cf := context.WithTimeout(ctx, syncTimeout)
+				updated, err := reader.Sync(timeoutCtx)
+				if err != nil {
+					lggr.Errorw("error syncing reader", "err", err)
+					cf()
```

Comment:
Again minor but this can be more idiomatic if you call a func that does `defer cancel()` e.g

```golang
case <-ticker:
  tick(syncTimeout, reader)

func tick(ctx context.Context, syncTimeout time.Duration, reader reader.CCIPReader) {
  ctx, cancel := context.WithTimeout(ctx, syncTimeout)
  defer cancel()

  // .. rest of code here, return as usual, ctx will be canceled
}
```